### PR TITLE
ISPN-4046 Support indexing and searching of null non-string properties

### PR DESCRIPTION
**Pull request by anistor**
jira: https://issues.jboss.org/browse/ISPN-4046

### CODE REVIEW

File: parent/pom.xml [hunk 1]
```diff
@@ -119,7 +119,7 @@
       <version.h2.driver-within-wildfly>1.3.173</version.h2.driver-within-wildfly>
       <version.hibernate.core>5.0.0.CR3</version.hibernate.core>
       <version.hibernate.entitymanager>${version.hibernate.core}</version.hibernate.entitymanager>
-      <version.hibernate.hql.parser>1.3.0.Alpha1</version.hibernate.hql.parser>
+      <version.hibernate.hql.parser>1.3.0-SNAPSHOT</version.hibernate.hql.parser>
```
Comment (gustavonalle) [inline parent/pom.xml]: why SNAPSHOT?

Comment (anistor) [inline parent/pom.xml]: Ooops, forgot I need to release a new HQL, 1.3.0.Alpha2. Will do that shortly.

File: bom/pom.xml [hunk 2]
```diff
@@ -67,7 +67,7 @@
       <version.gnu.getopt>1.0.13</version.gnu.getopt>
       <version.hibernate.javax.persistence>1.0.0.Final</version.hibernate.javax.persistence>
       <version.hibernate.core>5.0.0.CR3</version.hibernate.core>
-      <version.hibernate.search>5.5.0.Alpha1</version.hibernate.search>
+      <version.hibernate.search>5.5.0-SNAPSHOT</version.hibernate.search>
```
Comment (anistor) [inline bom/pom.xml]: A release of hsearch is also needed... Let's postpone this PR for after CR1.

**Comment by galderz** [comment]
Needs rebasing

File: integrationtests/all-embedded-query-it/src/test/java/org/infinispan/all/embeddedquery/QueryDslConditionsTest.java [hunk 3]
```diff
@@ -860,7 +859,7 @@ public void testIsNull3() throws Exception {
       assertEquals(3, list.get(0).getId());
    }
 
-   @Ignore(value = "https://hibernate.atlassian.net/browse/HSEARCH-1956")
+   @Test
```
Comment (danberindei) [inline integrationtests/all-embedded-query-it/src/test/java/org/infinispan/all/embeddedquery/QueryDslConditionsTest.java]: Do you need the explicit `@Test` annotations?

**Comment by danberindei** [comment]
Integrated, thanks Adrian!

**Pull request closed without merging**